### PR TITLE
fix(core/webidl): escape IDL code for fix suggestions

### LINKED ISSUE #2563: Suggested fix for WebIDL should write out entities
Right now, the suggested fix for IDL in the ReSpec Pill are being treated as HTML... meaning that `<Whatever>` in  `Promise<Whatever>` (or any generic) are being treated as a HTML tag. 

Actual:
```
[Exposed=(DedicatedWorker,Window), SecureContext]
interface SerialPort {
  Promise open(optional SerialOptions options = {});
};
```

Expected:

```
[Exposed=(DedicatedWorker,Window), SecureContext]
interface SerialPort {
  Promise&lt;void&gt; open(optional SerialOptions options = {});
};
```

### PR DESCRIPTION
**Pull request by saschanaz**
Closes #2563

Can't test error messages yet, so just a fix.

### CODE REVIEW

File: src/core/webidl.js [hunk 1]
```diff
@@ -365,8 +370,10 @@ export async function run() {
     let details = `<pre>${validation.context}</pre>`;
     if (validation.autofix) {
       validation.autofix();
+      const idlToFix = webidl2.write(astArray[validation.sourceName]);
+      const escaped = xmlEscape(idlToFix);
```
Comment (marcoscaceres) [inline src/core/webidl.js]: finally we get to use xmlEscape 🥳

Comment (marcoscaceres) [reply src/core/webidl.js]: XML IS NOT DEAD!

**Comment by saschanaz** [comment]
>Should we add a test?

Can't test an error message for now, no? We really should make a way to do this.

**Comment by marcoscaceres** [comment]
Oops, missed the PR message. All good :)

**Pull request merged**